ui/page/dcrdex/market.go

```diff
@@ -0,0 +1,1018 @@
+						Width:       cryptomaterial.WrapContent,
+						Height:      cryptomaterial.WrapContent,
+						Margin:      layout.Inset{Bottom: u20},
+						Orientation: layout.Vertical,
+					}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							return semiBoldLabelGrey3(pg.Theme, gtx, values.String(values.StrPrice))
+						}),
+						layout.Rigid(func(gtx C) D {
+							// TODO: Fetch rate for the base asset from DEX server and exchange rate from rate source!
+							rate := 200.0
+							tradeRate := 0.0034353
+							var lb cryptomaterial.Label
+							if rate == 0 {
+								lb = pg.Theme.Label(values.TextSize14, pg.Printer.Sprintf("%f", tradeRate))
+							} else {
+								lb = pg.Theme.Label(values.TextSize14, pg.Printer.Sprintf("%f (~ %s)", tradeRate, utils.FormatAsUSDString(pg.Printer, tradeRate*rate)))
+							}
+							lb.Font.Weight = font.SemiBold
+							return lb.Layout(gtx)
+						}),
+					)
+				}),
+				layout.Rigid(func(gtx C) D {
+					return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							return semiBoldLabelGrey3(pg.Theme, gtx, values.String(values.Str24hLow))
+						}),
+						layout.Rigid(func(gtx C) D {
+							lb := pg.Theme.Label(values.TextSize14, pg.Printer.Sprintf("%f", 0.0034353 /* TODO: use DEX server value */))
+							lb.Font.Weight = font.SemiBold
+							return lb.Layout(gtx)
+						}),
+					)
+				}),
+			)
+		}),
+		layout.Flexed(0.33, func(gtx C) D {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return cryptomaterial.LinearLayout{
+						Width:       cryptomaterial.WrapContent,
+						Height:      cryptomaterial.WrapContent,
+						Margin:      layout.Inset{Bottom: u20},
+						Orientation: layout.Vertical,
+					}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							return semiBoldLabelGrey3(pg.Theme, gtx, values.String(values.Str24hChange))
+						}),
+						layout.Rigid(func(gtx C) D {
+							// TODO: Use real values.
+							priceChange := 0.0010353
+							priceChangePercent := 0.18
+							lb := pg.Theme.Label(values.TextSize14, pg.Printer.Sprintf(`%f (%.2f`, priceChange, priceChangePercent)+"%)")
+							lb.Font.Weight = font.SemiBold
+							if priceChangePercent < 0 {
+								lb.Color = pg.Theme.Color.OrangeRipple
+							} else if priceChangePercent > 0 {
+								lb.Color = pg.Theme.Color.GreenText
+							}
+							return lb.Layout(gtx)
+						}),
+					)
+				}),
+				layout.Rigid(func(gtx C) D {
+					return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							return semiBoldLabelGrey3(pg.Theme, gtx, values.StringF(values.Str24hVolume, "DCR" /* TODO: use market base asset symbol */))
+						}),
+						layout.Rigid(func(gtx C) D {
+							lb := pg.Theme.Label(values.TextSize14, pg.Printer.Sprintf("%f", 4400.0477380 /* TODO: use DEX server value */))
+							lb.Font.Weight = font.SemiBold
+							return lb.Layout(gtx)
+						}),
+					)
+				}),
+			)
```

Comment:
This sets of codes are repeated. Move this into a wrapper method and call where needed passing the relevant data to it.